packages/react-native-app-auth/ios/RNAppAuth.m

```diff
@@ -499,35 +538,54 @@ - (void)endSessionWithConfiguration: (OIDServiceConfiguration *) configuration
+#endif
     if (![[appDelegate class] conformsToProtocol:@protocol(RNAppAuthAuthorizationFlowManager)]) {
         [NSException raise:@"RNAppAuth Missing protocol conformance"
                     format:@"%@ does not conform to RNAppAuthAuthorizationFlowManager", appDelegate];
     }
     appDelegate.authorizationFlowManagerDelegate = self;
     __weak typeof(self) weakSelf = self;
 
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
     rnAppAuthTaskId = [UIApplication.sharedApplication beginBackgroundTaskWithExpirationHandler:^{
         [UIApplication.sharedApplication endBackgroundTask:rnAppAuthTaskId];
         rnAppAuthTaskId = UIBackgroundTaskInvalid;
     }];
 
     UIViewController *presentingViewController = appDelegate.window.rootViewController.view.window ? appDelegate.window.rootViewController : appDelegate.window.rootViewController.presentedViewController;
+#elif TARGET_OS_OSX
+    NSWindow *presentingWindow = [NSApplication sharedApplication].mainWindow;
+    if(!presentingWindow){
+        reject(@"authentication_failed", @"Unable to get the main window to present an authorization request. Try authenticating again with the main window open.", nil);
+        return;
+    }
+#endif
 
 #if TARGET_OS_MACCATALYST
     id<OIDExternalUserAgent> externalUserAgent = nil;
 #elif TARGET_OS_IOS
     id<OIDExternalUserAgent> externalUserAgent = iosCustomBrowser != nil ? [self getCustomBrowser: iosCustomBrowser] : [self getExternalUserAgentWithPresentingViewController:presentingViewController
                                                                                                                                     prefersEphemeralSession:prefersEphemeralSession];
+#elif TARGET_OS_OSX
+    id<OIDExternalUserAgent> externalUserAgent = nil;
 #endif
     
     _currentSession = [OIDAuthorizationService presentEndSessionRequest: endSessionRequest
                                                       externalUserAgent: externalUserAgent
                                              callback: ^(OIDEndSessionResponse *_Nullable response, NSError *_Nullable error) {
                                                           typeof(self) strongSelf = weakSelf;
                                                           strongSelf->_currentSession = nil;
+#if TARGET_OS_IOS || TARGET_OS_MACCATALYST
                                                           [UIApplication.sharedApplication endBackgroundTask:rnAppAuthTaskId];
                                                           rnAppAuthTaskId = UIBackgroundTaskInvalid;
+#elif TARGET_OS_OSX
+                                                          // Brings this app to the foreground.
+                                                          [[NSRunningApplication currentApplication] activateWithOptions:(NSApplicationActivateAllWindows)];
+#endif
```

Comment:
macOS AppKit apps don't get backgrounded, so we don't need to declare background tasks (but do benefit from refocusing the app).